internal/clients/team4/judge.go

```diff
@@ -102,29 +125,45 @@ func (j *judge) HistoricalRetributionEnabled() bool {
 // CallPresidentElection is called by the judiciary to decide on power-transfer
 // COMPULSORY: decide when to call an election following relevant rulesInPlay if you wish
 func (j *judge) CallPresidentElection(monitoring shared.MonitorResult, turnsInPower int, allIslands []shared.ClientID) shared.ElectionSettings {
-	// example implementation calls an election if monitoring was performed and the result was negative
-	// or if the number of turnsInPower exceeds 3
+
 	var electionsettings = shared.ElectionSettings{
 		VotingMethod:  shared.Runoff,
 		IslandsToVote: allIslands,
 		HoldElection:  false,
 	}
+
+	// calculate whether the term has ended
+	termEnded := uint(turnsInPower) > j.parent.getTurnLength(shared.President)
+
+	j.parent.LocalVariableCache[rules.TermEnded] = rules.VariableValuePair{
+		VariableName: rules.TermEnded,
+		Values:       []float64{boolToFloat(termEnded)},
+	}
+
+	// try not holding election
+	j.parent.LocalVariableCache[rules.ElectionHeld] = rules.VariableValuePair{
+		VariableName: rules.ElectionHeld,
+		Values:       []float64{boolToFloat(false)},
+	}
+
+	compliantWithRules := j.parent.CheckCompliance(rules.ElectionHeld)
+
 	if monitoring.Performed && !monitoring.Result {
+		// If president didn't perform it's duties we can try to elect new president. Potentially decide based on trust or something
 		electionsettings.HoldElection = true
-	}
-	if turnsInPower >= 2 {
+	} else if compliantWithRules {
+		// If we are compliant with rules ie. we don't have to (are not obliged) hold election
+		electionsettings.HoldElection = false
+	} else {
+		// Now we know that we have to hold elections. We can check whether we want to hold election based on trust or something
 		electionsettings.HoldElection = true
 	}
+
 	return electionsettings
 }
 
 // DecideNextPresident returns the ID of chosen next President
 // OPTIONAL: override to manipulate the result of the election
-func (j *judge) DecideNextPresident(winner shared.ClientID) shared.ClientID {
-	// overloaded
-	j.logf("hello world %v", winner)
-	return id
-}
 
 func (j *judge) logf(format string, a ...interface{}) {
```

Comment:
ah--you're using `t` here to logf. I prefer you instead pass in a logger function. `t.Log` is a function pointer you can pass in the judge. Your judge can then take in any logger function it wants.

Reply:
This was in #369--I missed this part. Well, it works now so I won't block you, but I'll create an issue

Reply:
if I recall correcltly I had some problems with logger in testing, hence this approach.

Reply:
this is a very side-effecty way of testing unfortunately, but I don't want to block this becuase of that (since the change bringing this in was in #369). Tracking in #386